mesonbuild/interpreter/interpreter.py

```diff
@@ -664,6 +664,9 @@ def func_import(self, node: mparser.BaseNode, args: T.Tuple[str],
     @typed_pos_args('files', varargs=str)
     @noKwargs
     def func_files(self, node: mparser.FunctionNode, args: T.Tuple[T.List[str]], kwargs: 'TYPE_kwargs') -> T.List[mesonlib.File]:
+        for filename in args[0]:
+            if filename.endswith(' '):
+                raise MesonException(f'{filename!r} ends with a space. This is probably an error.', file=node.filename, lineno=node.lineno, colno=node.colno)
```

Comment:
Should we mention why it's an error, e.g. that it breaks on Windows?

Are we worried about files passed as strings to a function that internally calls source_strings_to_files, but isn't the meson.build DSL function `files()` itself?

Reply:
I wasn't sure how many details should be put in the error message. Having a fine name ending with a space will always be confusing, anyway.

Yes, that's probably a good idea to move the check to `source_strings_to_files`, and it prevents adding a loop, which is a good thing.